Packs/CommonScripts/Scripts/IsDomainInternal/IsDomainInternal.py

```diff
@@ -0,0 +1,93 @@
+from time import sleep
+
+import demistomock as demisto  # noqa: F401
+from CommonServerPython import *  # noqa: F401
+
+POLLING_TIME = 2  # Interval to wait in seconds when polling to check if indicator was created
+
+
+def check_domain_internal(domain, internal_domains):
+    parts = domain.split(".")
+    for i in range(len(parts), 0, -1):
+        sub = ".".join(parts[-i:])
+        if sub in internal_domains:
+            return True
+    return False
+
+
+def create_markdown_table(domain_list):
+    domain_list = sorted(domain_list, key=lambda x: not x['Internal'])
+    markdown_string = "| Domain Name | Internal |\n| --- | --- |\n"
+    for domain in domain_list:
+        markdown_string += f"| {domain['Name']} | {domain['Internal']} |\n"
+    return markdown_string
+
+
+internal_domains_list = demisto.args().get("InternalDomainsListName", "InternalDomains")
+domains_to_check = argToList(demisto.args().get("Domains", None))
+
+# Get the list of internal domains from the XSOAR list:
+internal_domains = demisto.executeCommand("getList", {"listName": internal_domains_list})[0]['Contents']
+if "Item not found" in internal_domains:
+    return_error(f"The list name {internal_domains_list} does not exist.")
+
+# Split internal domains from XSOAR list to be a list:
+if internal_domains:
+    try:
+        internal_domains = internal_domains.split("\n")
+    except Exception as ex:
+        return_error(f"Could not parse the internal domains list. Please make sure that the list contains domain names, separated by new lines.\nThe exact error is: {ex}")
+else:
+    demisto.results("No internal domains were specified.")
+    quit()
+
+
+# Create list of domain names with internal property
+domain_list = [{"Name": domain, "Internal": check_domain_internal(domain, internal_domains)} for domain in domains_to_check]
+
+
+for domain in domain_list:
+    args_exists_check = {
+        "indicator": domain.get("Name")
+    }
+
+    args_create_or_set_indicator = {
+        "value": domain.get("Name"),
+        "type": "Domain",
+        "internal": domain.get("Internal")
+    }
+
+    # Check if indicator exists:
+    is_exist_res = demisto.executeCommand("CheckIndicatorValue", args_exists_check)
+    indicator_exists = is_exist_res[0].get("Contents", {})[0].get("Exists")
```

Comment:
default is `{}` (not list), but then the script accesses `[0]` - is `0` a dictionary key?

Reply:
That's actually a list:
![image](https://user-images.githubusercontent.com/43602124/217239195-025f7853-e361-4e6d-b25e-d5728c691628.png)

I guess I should do `get("Contents", [])`?

Reply:
Changed it to [], LMK if that's ok

Reply:
1. if `is_exist_res` is empty, we'll get an IndexError trying to access `[0]`
2. if `is_exist_res[0]` doesn't have `Contents`, we'll get `{}[0].get("Exists")` - IndexError

Reply:
Correct, but it will always have a value at [0] because that's the response we get back from server - regardless of whether it actually exists or not (I tested both cases). Any suggestion on what I should change?